gateway/src/lib.rs

```diff
@@ -101,6 +121,12 @@ pub mod pallet {
                     frame_system::RawOrigin<<Self as frame_system::Config>::AccountId>,
```

Comment:
```suggestion
                    frame_system::RawOrigin<T::AccountId>,
```
This should work because you have not defined your own `AccountId`